### PR TITLE
fix(volunteer): the problem of not creating events

### PR DESCRIPTION
**Pull request by ardasnturk**
- added `filterGroupDropDownData` to `useEffect` to ensure dropdown data is refreshed

HDVT-93

### CODE REVIEW

File: src/components/volunteer/VolunteerFormCalendar.tsx [hunk 1]
```diff
@@ -167,6 +167,10 @@ export const VolunteerFormCalendar = ({
       }, 300);
   }, [isRefetchingGroupsMy]);
 
+  useEffect(() => {
+    filterGroupDropDownData();
+  }, [filterGroupDropDownData]);
```
Comment (donni106) [inline src/components/volunteer/VolunteerFormCalendar.tsx:172]: why is this necessary now?
and wouldn't adding the dependency to the previous effect have the same result?

Comment (ardasnturk) [reply src/components/volunteer/VolunteerFormCalendar.tsx:172]: The `VolunteerFormCalendar` screen is logically similar to the `VolunteerFormConversation` screen. Both pages have a dropdown list. Through this list, users can select their groups. However, the dropdown list showing the groups on the `VolunteerFormCalendar` page cannot be loaded when the application is first opened. To solve this problem, I looked at the `VolunteerFormConversation` screen and added the `useEffect` to make the code similar to each other. This way the user can see the dropdown list when they go to the event creation page.

`VolunteerFormConversation`:
image

Comment (donni106) [reply src/components/volunteer/VolunteerFormCalendar.tsx:172]: and what about the question about the previous effect? they are very similar. can we remove the existing one after adding the new one?

Comment (ardasnturk) [reply src/components/volunteer/VolunteerFormCalendar.tsx:172]: Are you talking about this `useEffect`?
```
  useEffect(() => {
    !isRefetchingGroupsMy &&
      setTimeout(() => {
        filterGroupDropDownData();
      }, 300);
  }, [isRefetchingGroupsMy]);
```

I still think that this effect should remain. I think that if we delete it, the groups may not be refreshed properly.

Comment (donni106) [reply src/components/volunteer/VolunteerFormCalendar.tsx:172]: both effects are running the same method, right? so at least we should combine them?

Comment (ardasnturk) [reply src/components/volunteer/VolunteerFormCalendar.tsx:172]: I think we can update the code in this way.
```
  useEffect(() => {
    !isRefetchingGroupsMy &&
      setTimeout(() => {
        filterGroupDropDownData();
      }, 300);
  }, [isRefetchingGroupsMy, filterGroupDropDownData]);
```

Comment (donni106) [reply src/components/volunteer/VolunteerFormCalendar.tsx:172]: i agree

**Review by donni106 — APPROVED** [review]

**Review by d4v3d4vE — APPROVED** [review]

File: src/components/volunteer/VolunteerFormCalendar.tsx (hunk 1 above)
Comment (d4v3d4vE) [reply src/components/volunteer/VolunteerFormCalendar.tsx]: thanks for this explanation/deconstruction you guy did here :)
